packages/big-design/src/components/List/styled.tsx

```diff
@@ -13,6 +13,7 @@ export const StyledList = styled.ul<Partial<ListProps<unknown>>>`
   outline: none;
   overflow-y: auto;
   padding: ${({ theme }) => theme.spacing.xSmall} 0;
+  position: fixed;
```

Comment:
I did notice that this breaks the `ButtonGroup` component (screenshot attached). I'm starting to think this might be an issue with how the checkbox is built and I am considering maybe that's something we need to look at instead.
![Screenshot 2024-06-04 at 15 08 50](https://github.com/bigcommerce/big-design/assets/10539418/06289035-9598-42ae-b8a0-75dfd4721762)

Reply:
Ah good spot!

Reply:
I may be spraying and praying but how about position fixed on the checkbox instead? That also seems to remove the scroll with the checkbox remaining hidden. That'll be a more localised change so less likely to affect other components.

Reply:
I believe the `position:fixed` route will cause some accessibility issues if I am not mistaken. However, I did a little digging and I found a quick and dirty solution to fix it for the `MultiSelect`:

```diff
diff --git a/packages/big-design/src/components/List/styled.tsx b/packages/big-design/src/components/List/styled.tsx
index b821277d..59f2c93a 100644
--- a/packages/big-design/src/components/List/styled.tsx
+++ b/packages/big-design/src/components/List/styled.tsx
@@ -4,8 +4,6 @@ import styled from 'styled-components';
 import { ListProps } from './List';
 
 export const StyledList = styled.ul<Partial<ListProps<unknown>>>`
-  ${({ theme }) => theme.shadow.raised};
-
   background-color: ${({ theme }) => theme.colors.white};
   color: ${({ theme }) => theme.colors.secondary70};
   margin: 0;
@@ -13,7 +11,6 @@ export const StyledList = styled.ul<Partial<ListProps<unknown>>>`
   outline: none;
   overflow-y: auto;
   padding: ${({ theme }) => theme.spacing.xSmall} 0;
-  position: fixed;
   z-index: ${({ theme }) => theme.zIndex.popover};
 `;
 
diff --git a/packages/big-design/src/components/MultiSelect/MultiSelect.tsx b/packages/big-design/src/components/MultiSelect/MultiSelect.tsx
index 54e0ccc7..4d154a63 100644
--- a/packages/big-design/src/components/MultiSelect/MultiSelect.tsx
+++ b/packages/big-design/src/components/MultiSelect/MultiSelect.tsx
@@ -465,22 +465,24 @@ export const MultiSelect = typedMemo(
         {renderLabel}
         {renderInput}
         <Box ref={popperRef} style={styles.popper} {...attributes.poppper} zIndex="popover">
-          <List
-            {...ariaLabelledBy}
-            action={action}
-            addItem={addSelectedItem}
-            autoWidth={autoWidth}
-            filteredItems={filteredOptions}
-            getItemProps={getItemProps}
-            getMenuProps={getMenuProps}
-            highlightedIndex={highlightedIndex}
-            isOpen={isOpen}
-            items={options}
-            maxHeight={maxHeight}
-            removeItem={removeItem}
-            selectedItems={selectedOptions}
-            update={update}
-          />
+          <Box style={{ position: 'relative', overflow: 'hidden', height: '100%' }} shadow="raised">
+            <List
+              {...ariaLabelledBy}
+              action={action}
+              addItem={addSelectedItem}
+              autoWidth={autoWidth}
+              filteredItems={filteredOptions}
+              getItemProps={getItemProps}
+              getMenuProps={getMenuProps}
+              highlightedIndex={highlightedIndex}
+              isOpen={isOpen}
+              items={options}
+              maxHeight={maxHeight}
+              removeItem={removeItem}
+              selectedItems={selectedOptions}
+              update={update}
+            />
+          </Box>
         </Box>
       </div>
     );
```

I would create a new styled-component for the new `Box` wrapper, otherwise that seems to work. Could you fact check that though?

Reply:
You can def take credit for this if you want to push this change to this PR 😉

Reply:
😄 Ha thanks. I've pushed an update. I ran into one issue which was the suggested removal of the shadow on the List component

```diff
diff --git a/packages/big-design/src/components/List/styled.tsx b/packages/big-design/src/components/List/styled.tsx
index b821277d..59f2c93a 100644
--- a/packages/big-design/src/components/List/styled.tsx
+++ b/packages/big-design/src/components/List/styled.tsx
@@ -4,8 +4,6 @@ import styled from 'styled-components';
 import { ListProps } from './List';
 
 export const StyledList = styled.ul<Partial<ListProps<unknown>>>`
-  ${({ theme }) => theme.shadow.raised};
-
```

This appeared to cause a regression on other components, e.g. Dropdown, Select etc - they no longer had a drop shadow.

I've taken the approach of keeping it on the List component and adding the additional shadow on the new container for the MultiSelect. 

I believe the alternative would to remove the shadow from List and then add a new raised box to all current usages of List. Let me know if you'd prefer that.

Reply:
Ahh good catch!

I did a little testing this morning and I think we can create this overflow wrapper inside the `List` component and have it apply to all components so we don't have to individually apply it to the other usages of `List`. I couldn't break any of the components using the `List` component so I think we should be good. Let me know what you think though:

```diff
diff --git a/packages/big-design/src/components/List/List.tsx b/packages/big-design/src/components/List/List.tsx
index 8b96bed0..e61f56b2 100644
--- a/packages/big-design/src/components/List/List.tsx
+++ b/packages/big-design/src/components/List/List.tsx
@@ -20,7 +20,7 @@ import { SelectAction, SelectOption, SelectOptionGroup, SelectProps } from '../S
 import { ListGroupHeader } from './GroupHeader';
 import { ListGroupSeparator } from './GroupSeparator';
 import { ListItem } from './Item';
-import { StyledList } from './styled';
+import { StyledList, StyledListOverflowWrapper } from './styled';
 
 export interface ListProps<T> extends ComponentPropsWithoutRef<'ul'> {
   action?: SelectAction;
@@ -208,26 +208,28 @@ const StyleableList = typedMemo(
     }, [action, items, renderAction, renderGroup, renderItems]);
 
     return (
-      <StyledList
-        {...getMenuProps({
-          ...props,
-          onKeyDown: (event) => {
-            if (event.key === 'Enter') {
-              const element = event.currentTarget.children[highlightedIndex];
-              const link = element.querySelector('a');
+      <StyledListOverflowWrapper>
+        <StyledList
+          {...getMenuProps({
+            ...props,
+            onKeyDown: (event) => {
+              if (event.key === 'Enter') {
+                const element = event.currentTarget.children[highlightedIndex];
+                const link = element.querySelector('a');
 
-              // We want to click the link if it is selected
-              if (link && !link.getAttribute('disabled')) {
-                link.click();
+                // We want to click the link if it is selected
+                if (link && !link.getAttribute('disabled')) {
+                  link.click();
+                }
               }
-            }
-          },
-          ref: forwardedRef,
-        })}
-        maxHeight={maxHeight}
-      >
-        {isOpen && renderChildren}
-      </StyledList>
+            },
+            ref: forwardedRef,
+          })}
+          maxHeight={maxHeight}
+        >
+          {isOpen && renderChildren}
+        </StyledList>
+      </StyledListOverflowWrapper>
     );
   },
 );
diff --git a/packages/big-design/src/components/List/styled.tsx b/packages/big-design/src/components/List/styled.tsx
index b821277d..e9b7112e 100644
--- a/packages/big-design/src/components/List/styled.tsx
+++ b/packages/big-design/src/components/List/styled.tsx
@@ -3,9 +3,15 @@ import styled from 'styled-components';
 
 import { ListProps } from './List';
 
-export const StyledList = styled.ul<Partial<ListProps<unknown>>>`
-  ${({ theme }) => theme.shadow.raised};
+export const StyledListOverflowWrapper = styled.div`
+  ${({ theme }) => theme.shadow.raised}
+
+  position: relative;
+  overflow: hidden;
+  height: 100%;
+`;
 
+export const StyledList = styled.ul<Partial<ListProps<unknown>>>`
   background-color: ${({ theme }) => theme.colors.white};
   color: ${({ theme }) => theme.colors.secondary70};
   margin: 0;
@@ -13,7 +19,6 @@ export const StyledList = styled.ul<Partial<ListProps<unknown>>>`
   outline: none;
   overflow-y: auto;
   padding: ${({ theme }) => theme.spacing.xSmall} 0;
-  position: fixed;
   z-index: ${({ theme }) => theme.zIndex.popover};
 `;
 ```

Reply:
Even better @chanceaclark! That works from my testing. Pushed as an update.